src/recur.c

```diff
@@ -811,6 +811,17 @@ recur_item_find_occurrence(time_t item_start, long item_dur,
 	default:
 		EXIT(_("unknown item type"));
 	}
+	/* Impossible days must be ignored (according to RFC 5545). */
+	if (!ISLEAP(1900 + lt_item_day.tm_year) &&
+	    lt_item_day.tm_mon == 1 && lt_item_day.tm_mday == 29)
+		return 0;
+	if (lt_item_day.tm_mon == 1 &&
+	    (lt_item_day.tm_mday == 30 || lt_item_day.tm_mday == 31))
+		return 0;
+	if ((lt_item_day.tm_mon == 3 || lt_item_day.tm_mon == 5 ||
+	     lt_item_day.tm_mon == 8 || lt_item_day.tm_mon == 10) &&
+	    lt_item_day.tm_mday == 31)
+		return 0;
```

Comment:
Any reason you are not using check_date() here?

Reply:
I guess something like returning 0 if `(rpt_type == RECUR_MONTHLY || rpt_type == RECUR_YEARLY) && !check_date())` (untested) should work.

Reply:
Or maybe not, given the way we manipulate the months above. But that makes we wonder whether your check actually works? What happens when using a very high frequency? Say a monthly item starting on February 29th on a leap year and having a frequency of 24 months? Would your check still work?

Reply:
I had quite forgotten `check_date()` to be honest. But neither my check nor `check_date()` can work without some tweaking of `lt_item_day.tm_year` and `lt_item_day.tm_mon`. The latter may be negative as you indicated.

Any check should also be limited to the yearly and monthly cases (as weekly and daily lead to no ambiguous dates).

Another thing: I added a test case to the existing `apts-recur` data file. Would you prefer a separate data file?

Reply:
What should work is converting the computed `time_t` back to a `struct tm` and verify that the `tm_mday` matches the original `tm_mday`.

Reusing the existing data files in the tests is fine if it does not break any other test cases.

Reply:
Indeed! A direct verification is much better.

I have added a test case for yearly and uploaded the changes.